### PR TITLE
picker(live_grep): add option to grep only over open files

### PR DESCRIPTION
**Pull request by jameswalmsley**
No idea if I've implemented this correctly at all.. however it works.

I wanted to be able to grep just the currently open files, and then put
the results into a quick-list.

All feedback appreciated, and happy to refactor as required.

### CODE REVIEW

**Comment by elianiva** [comment]
thanks, but I'm pretty sure we already have this feature. It's called `current_buffer_fuzzy`

see https://github.com/nvim-telescope/telescope.nvim/wiki/Showcase#current-buffer-fuzzy-find

**Comment by jameswalmsley** [comment]
@elianiva Thanks for the response. current_buffer_fuzzy only does a fuzzy on the current buffer.
live_grep_open_files allows you to grep all files that are currently open in nvim.

current_buffer_fuzzy also doesn't seem to support sending a list of items to the qflist.

I think the advantages are:

  * Matches across all files currently open in nvim
  * grep i.e. full regular expressions matching
  * Send a selection or all results to qflist.

This scenario can be really powerful when you are rebasing a large code base, and need to make a large number of identical changes to all files in a patch.

**Comment by elianiva** [comment]
ah, I see. I thought current open file is the *current* file you're editing. I misread the *files* as *file*.

**Comment by elianiva** [comment]
on second thought, I think it would be better to make the current live_grep have an option to do this instead of creating a new picker. Maybe an option like `current_open_files=true` then it will only grep on the current open files.

**Comment by jameswalmsley** [comment]
@elianiva I like that idea. I'll update my PR.

**Comment by jameswalmsley** [comment]
@elianiva I've updated it to an option (grep_open_files=true) 
Can you think of a better name, (restrict_open=true) ?

Linting seems to be failing on some code I didn't change.

**Comment by elianiva** [comment]
don't worry about the lint, https://github.com/nvim-telescope/telescope.nvim/pull/676 takes care of that. Just need to wait for docgen to be fixed, otherwise if I merge that PR now, our doc will be gone 😆 

`grep_open_files` sounds good to me

**Review by elianiva — APPROVED** [review]

File: lua/telescope/builtin/init.lua
Comment (elianiva) [inline lua/telescope/builtin/init.lua:26]: oh, you forgot to remove this :)

File: lua/telescope/builtin/files.lua
Comment (elianiva) [inline lua/telescope/builtin/files.lua:55]: ..and I'm pretty sure this `not grep_open_files` isn't needed.
also you can do `vim.api.nvim_buf_get_name(bufnr)` instead of this URI stuff. 

I pressed approve to soon apparently :laughing: Could you fix this as well? :)

Comment (jameswalmsley) [reply lua/telescope/builtin/files.lua]: Nice, I've updated it to use the nvim api.
Also removed the extra `not grep_open_files`.. I was kind of thinking we don't want to iterate the search dirs.

But a user shouldn't provide both options anyway.

**Comment by jameswalmsley** [comment]
@elianiva I've rebased this branch onto the latest master.
Do you know if I can get this merged?

**Comment by elianiva** [comment]
I like it so far, will have to test it again tomorrow. We should ask someone else before merging it, I don't feel comfortable merging stuff on my own :laughing:

@tami5 thoughts?

**Comment by tami5** [comment]
It looks great to me, I'm going to test out in few hours.

**Comment by elianiva** [comment]
LGTM! Just tested it out and it works, Tami can merge if he wants.

**Comment by tami5** [comment]
@jameswalmsley LGTM great work. It would be awesome to do document this in the source code and readme as you see fit so others can use it if the like to.

**Comment by jameswalmsley** [comment]
@elianiva @tami5 Many thanks for the feedback, I'll add some documentation today. :)

**Comment by jameswalmsley** [comment]
@tami5 @elianiva I've added some option documentation for live_grep..
Wasn't sure exactly where options for pickers are documented in the README, so I've added a small table for live_grep under the picker documentation.

**Comment by Conni2461** [comment]
@tami5 fyi we actually wanted to trim down the documentation in readme. We NEVER included options in the readme because if he had we would have ended up with a > 1000 lines and nobody finds anything. Here is an example https://github.com/nvim-telescope/telescope.nvim/pull/659#discussion_r596265905 on how `opts` should be documented post docgen. Nobody has just started work on it.

I will still be here for questions so feel free to ping me :)

**Comment by tami5** [comment]
> @tami5 fyi we actually wanted to trim down the documentation in readme. We NEVER included options in the readme because if he had we would have ended up with a > 1000 lines and nobody finds anything. Here is an example https://github.com/nvim-telescope/telescope.nvim/pull/659#discussion_r596265905 on how `opts` should be documented post docgen. Nobody has just started work on it.
> 
> 
> 
> I will still be here for questions so feel free to ping me :)

Oh my bad conni.

**Pull request merged**
